### PR TITLE
Add terraform whitelists

### PR DESCRIPTION
**Pull request by luigibk**
Resolves # . (This is applicable only if this pull request relates to a GitHub issue, delete the line otherwise)

[[AB#467](https://dev.azure.com/hmcts/90472b63-dc8d-4ae6-8819-a32bd211e914/_workitems/edit/467)]

### CODE REVIEW

File: terraform-infra-approvals/README.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# Terraform  whitelists folder
+
+A whitelist is a list of Terraform resources and modules used to enforce constraints on infrastructure creation. 
+Whitelist files can be global, in which case it is named `global.json` or specific to a github repo and named `<repo-name>.json`.
```
Comment (timja) [inline terraform-infra-approvals/README.md]: how is this going to work in the jenkins library? 
are we going to try load a file for every repo?
or someone has to send a PR to the jenkins-library or add a flag on every repo?

Also this doesn't scale with CODEOWNERS very well you would either need to add an entry for each new json file or *.json but *.json isn't really appropriate as this repo is used for other things?

could we just have the specific github repo one as one file, with another key added to do it to say which repo it's for? I get that you wouldn't be able to pass it directly to the tool, without pre-processing, but that would be a simple jq command?

Comment (luigibk) [reply terraform-infra-approvals/README.md]: If a repo does not have any "special approvals" there should be no file there. In fact there should be very few repos that need that. I have already discussed this with @njrich28 and that is the solution we have agreed.

Comment (timja) [reply terraform-infra-approvals/README.md]: Could you do a corresponding PR to the jenkins-library first please, also you didn't answer my question about CODEOWNERS. or how we will know to check for this file

Comment (njrich28) [reply terraform-infra-approvals/README.md]: devops (or prod-infra-approvals) are CODEOWNERS for every file in this directory

Comment (timja) [reply terraform-infra-approvals/README.md]: not atm 😉

File: terraform-infra-approvals/README.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
+# Terraform  whitelists folder
```
Comment (timja) [inline terraform-infra-approvals/README.md]:
```suggestion
# Terraform  whitelists
```

**Pull request merged**